share/p2p/shrexnd/server.go

```diff
@@ -95,90 +140,65 @@ func (srv *Server) handleNamespacedData(ctx context.Context, stream network.Stre
 	var req pb.GetSharesByNamespaceRequest
 	_, err = serde.Read(stream, &req)
 	if err != nil {
-		logger.Warnw("server: reading request", "err", err)
-		stream.Reset() //nolint:errcheck
-		return
+		return nil, fmt.Errorf("reading request: %w", err)
```

Comment:
why remove the stream.Reset?

Reply:
it is not removed, it is is extracted to single common place:
```
func (srv *Server) handler(ctx context.Context) network.StreamHandler {
	return func(s network.Stream) {
		err := srv.handleNamespacedData(ctx, s)
		if err != nil {
			s.Reset() //nolint:errcheck
			return
		}
		if err = s.Close(); err != nil {
			log.Debugw("server: closing stream", "err", err)
		}
	}
}
```
It allowed to remove 5 copies of s.Reset() //nolint:errcheck